spec/datadog/appsec/waf_spec.rb

```diff
@@ -202,6 +202,41 @@
         end
       end
     end
+
+    context 'stress testing' do
+      let(:run_count) { 500 }
+      let(:thread_count) { 200 }
+
+      it 'creates a context in each thread' do
+        handle
+
+        result = { ok: 0, match: 0, err_internal: 0, err_invalid_object: 0, err_invalid_argument: 0 }
+        start_barrier = Barrier.new(thread_count)
+        res_mutex = Mutex.new
+        threads = []
+
+        thread_count.times do
+          threads << Thread.new do
```

Comment:
```suggestion

        threads = thread_count.times.map do
          Thread.new do
```

Reply:
Fixed in b4f21dc9fb679c464bd9eea3928fcf8f7046dd1e